### ISSUE TITLE
Basic auth in index-url for pip fails rule

### ISSUE THREAD

**Issue by bneijt** [hadolint/hadolint#452]
- [x] This is a bug report
- [x] I searched existing issues before opening this one

### Expected behavior

   pip install --index-url https://user:pass@example.com arrow==10

should be valid as I pin the version.

### Actual behavior

Rule `DL3013` is triggered.

### Steps to reproduce the behavior

Create a dockerfile with

  RUN pip install --index-url https://user:pass@example.com arrow==1.0.0

Execute hadolint on the docker file.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
% stack run -- hadolint --version                                                                                                                                                                         
Haskell Dockerfile Linter v1.18.0-6-ga0d655d
```

**Additional environment details (OS, stack version, etc.)**
Linux,
stack Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (dirty) (8103 commits) x86_64

Also reproducable in spec test:
```haskell
            it "pip install --index-url with --extra-index-url with basic auth" $ do
                ruleCatchesNot
                    pipVersionPinned
                    "RUN pip install --index-url https://user:pass@eg.com/foo --extra-index-url https://user:pass@ex-eg.io/foo foobar==1.0.0"
                onBuildRuleCatchesNot
                    pipVersionPinned
                    "RUN pip install --index-url https://user:pass@eg.com/foo --extra-index-url https://user:pass@ex-eg.io/foo foobar==1.0.0"
```

**Comment by davidhart82** [comment]
Can confirm that we see this too.
